extensions/table.c

```diff
@@ -282,7 +282,7 @@ static cmark_node *try_opening_table_header(cmark_syntax_extension *self,
   set_n_table_columns(parent_container, header_row->n_columns);
 
   uint8_t *alignments =
-      (uint8_t *)parser->mem->calloc(header_row->n_columns, sizeof(uint8_t));
+      (uint8_t *)parser->mem->calloc(marker_row->n_columns, sizeof(uint8_t));
```

Comment:
Since the number of alignments can now exceed the amount of header columns, this makes the alignments array's size based on the marker row columns directly.